1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,8 +1,46 @@
-function setAlarm() {}
+let remainingTime;
+let intervalId;
+let isPaused = false;
+
+function setAlarm() {
+  const alarmSetValue = document.getElementById("alarmSet").value;
+  remainingTime = alarmSetValue;
+  let minutes = Math.floor(remainingTime / 60);
+  let seconds = remainingTime % 60;
+  document.getElementById("timeRemaining").innerHTML = `Time Remaining: ${
+    minutes < 10 ? "0" : ""
+  }${minutes}:${seconds < 10 ? "0" : ""}${seconds}`;
+  intervalId = setInterval(countdown, 1000);
+}
```

Comment:
Hi Mo, good job . just maybe it is a good idea to search about (button.disabled) .click on **set alarm button** several times and see what happen.